Automattic/jetpack
# Search: Add clear search button and polish surrounding visuals

Fixes #18879.

#### Changes proposed in this Pull Request:
* Adds a `clear` button adjacent to the search input area.
* Adds a visible border between the search input area and the close overlay button.
* Refactors the DOM hierarchy for SearchResults, dark theme styles.

> <img width="1697" alt="Screen Shot 2021-03-02 at 1 09 45 PM" src="https://user-images.githubusercontent.com/4044428/109708390-96118080-7b58-11eb-8952-eb6f22443f7b.png">
> <img width="1717" alt="Screen Shot 2021-03-02 at 12 59 43 PM" src="https://user-images.githubusercontent.com/4044428/109708396-99a50780-7b58-11eb-8c44-824ae73f67e5.png">


#### Jetpack product discussion
See #18879.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your site with a Jetpack Search subscription.
* Perform a site search. Ensure that the overlay appears with the new `clear` button adjacent to the search input.
* Click the `clear` button. Ensure that it clears the search input.

#### Proposed changelog entry for your changes:
* None.
